File: backend/src/main/java/com/elbuensabor/entities/Cliente.java
package com.elbuensabor.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "cliente")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Cliente extends Base {
    @Column
    private String nombre;
    @Column
    private String apellido;
    @Column
    private Long telefono;
    @Column
    private String email;

    @OneToOne
    @JoinColumn(name = "usuario_id")
    private Usuario usuario;

    @OneToMany(mappedBy = "cliente")
    private List<Pedido> pedidos = new ArrayList<Pedido>();

    @OneToMany(mappedBy = "cliente")
    private List<Domicilio> domicilios = new ArrayList<Domicilio>();

}

File: frontend/el-buen-sabor-frontend/src/app/entidades/Cliente.ts
import { Domicilio } from './Domicilio';
import { Pedido } from './Pedido';
import { Usuario } from './Usuario';

export class Cliente {
  id: number = 0;
  nombre: string = '';
  apellido: string = '';
  telefono: number = 0;
  email: string = '';
  usuario!: Usuario; // remove the instantiation here
  pedidos: Pedido[] = [];
  domicilios: Domicilio[] = [];

  // constructor(usuario: Usuario) { // pass the needed instance through the constructor
  //   this.usuario = usuario;
  // }
}